### DISCUSSION TITLE
TF footprinting Analysis and Interpretation

### DISCUSSION THREAD

**Discussion by IdoMeshulam** [stuart-lab/signac#968] [Q&A] [answered]
Hello,

I am interested in learning more on how to address the TF footprinting workflow, and wanted to raise three main questions. Attached is a screenshot from the Signac Transcription Factor Footprinting vignette.

![Screenshot 2022-02-10 121629](https://user-images.githubusercontent.com/80117279/153386286-88b781ff-8d32-4233-b19c-edbdbf8f0df7.jpeg)

1) I want to thoroughly understand how to interpret the footprinting plots. From the Signac preprint https://www.biorxiv.org/content/10.1101/2020.11.09.373613v1 under section "Motif footprinting" I understand that the plot represents the difference between the expected Tn5 insertion rate at each 6 nucleotide sequence (based on natural frequency of occurrence in the genome) compared to the actual Tn5 insertion rate. How can this plot be interpreted, biologically? When is there reason to assume that there is an active transcription factor binded to the motif, based on the plot? Are there any guidelines on how to interpret the plot based on the peak-dip-peak pattern or other patterns that may occur, and to understand whether it is significant or not?

2) The second question stems very much from the first - I'm looking for a more quantitative approach to analyze footprints rather than a qualitative one. Is there a known vignette or pipeline I can use to quantitively analyze the footprints to yield more solid, scientifically valid conclusions, based on calculations rather than on qualitative analysis by "looking at the pattern in the plot"?

3) Thirdly, as a consequence of the second question, is there any way to "scan" through a list of motifs to automatically search for the "peak-dip-peak" footprinting pattern, or any other "interesting" patterns that show a statistically significant difference of the expected vs actual Tn5 insertion rate? Currently in the TF vignette https://satijalab.org/signac/articles/footprint.html I can only run a footprint of a motif and then manually see how the plot looks like, rather than scanning all known motifs to automatically find these patterns beforehand.

Thank you so much for this awesome tool!

**Comment by timoast** [comment, marked as answer]
You can check this paper for more explanation of the footprinting method: https://www.science.org/doi/10.1126/science.aav1898

> I want to thoroughly understand how to interpret the footprinting plots. From the Signac preprint https://www.biorxiv.org/content/10.1101/2020.11.09.373613v1 under section "Motif footprinting" I understand that the plot represents the difference between the expected Tn5 insertion rate at each 6 nucleotide sequence (based on natural frequency of occurrence in the genome) compared to the actual Tn5 insertion rate. How can this plot be interpreted, biologically? When is there reason to assume that there is an active transcription factor binded to the motif, based on the plot? Are there any guidelines on how to interpret the plot based on the peak-dip-peak pattern or other patterns that may occur, and to understand whether it is significant or not?

The important thing to look at is the height of the motif-flanking region. I would not really interpret the dip in the center to mean anything, since it is influenced a lot by the Tn5 insertion bias. Biologically, higher counts flanking the motif positions means that more Tn5 insertions were observed surrounding that motif site in those cells.

> The second question stems very much from the first - I'm looking for a more quantitative approach to analyze footprints rather than a qualitative one. Is there a known vignette or pipeline I can use to quantitively analyze the footprints to yield more solid, scientifically valid conclusions, based on calculations rather than on qualitative analysis by "looking at the pattern in the plot"?

You can compute the height of the motif flank. We don't currently provide a function to do this in Signac but you can extract the footprinting matrix using the `GetFootprintData()` function, and there is some code in `PlotFootprint()` you could look at to see how to get the motif flanks.

> Thirdly, as a consequence of the second question, is there any way to "scan" through a list of motifs to automatically search for the "peak-dip-peak" footprinting pattern, or any other "interesting" patterns that show a statistically significant difference of the expected vs actual Tn5 insertion rate? Currently in the TF vignette https://satijalab.org/signac/articles/footprint.html I can only run a footprint of a motif and then manually see how the plot looks like, rather than scanning all known motifs to automatically find these patterns beforehand.

No, computing the motif footprint is quite intensive since you have to quantify a lot of genomic regions. For a more discovery-based analysis you could run chromVAR and find motifs that deviate in their accessibility, and then follow up by running footprinting on the top motifs.

**Reply by IdoMeshulam** [reply]
Thank you very much for the detailed response!

Why do you think that the height of the motif flanking region is the most interesting thing to look at, and not interpret any meaning to the dip in the center? If many Tn5 insertions were observed surrounding a specific motif site, but a strong dip exists on the motif, can't that possibly suggest that a certain transcription factor or other protein may be currently bounded to that motif, thus preventing insertion by the Tn5 at that specific locus? 

If not, what other ways could I have using Signac to look for signs of active transcription currently occurring or not in the sample, and at which locations in the genome?

**Reply by timoast** [reply]
> If many Tn5 insertions were observed surrounding a specific motif site, but a strong dip exists on the motif, can't that possibly suggest that a certain transcription factor or other protein may be currently bounded to that motif, thus preventing insertion by the Tn5 at that specific locus?

The idea of footprinting was (as far as I know) first developed for DNase-seq, and the idea was exactly as you say, to identify TF binding sites by the blocking of DNase activity at that site. However for scATAC-seq, there are two problems: 1) the signal is very sparse; 2) Tn5 has substantial insertion sequence bias. This is difficult to overcome, and that's why I don't like to try to interpret the dip in the center too much. The simplest explanation is Tn5 insertion bias. However, the height of the flank can tell you if the region around those motifs is open, which often corresponds to binding of the TF.

> If not, what other ways could I have using Signac to look for signs of active transcription currently occurring or not in the sample, and at which locations in the genome?

You mean TF binding? Or actual transcription? For TF binding, the height of the footprinting flank is informative, and you could try some more complex analysis integrating with scRNA and see if the TF is expressed, and try something like SCENIC: https://www.nature.com/articles/nmeth.4463

For actual transcription, scATAC is probably not the right assay, you may as well do scRNA-seq or multiome

**Reply by IdoMeshulam** [reply]
Okay, I understand. I'll do some more reading into the subject.
Thanks again for all the help :)
